sktime/markov/msm/augmented_msm.py

```diff
@@ -0,0 +1,623 @@
+            frac *= 0.1
+
+            if frac < 1e-12:
+                self._log.info("Small gradient fraction")
+                break
+
+            state.d_m_hat = state.m_hat - mhat_old
+            state.log_likelihood_prev = float(_ll_new)
+
+        state.log_likelihoods.append(_ll_new)
+
+    def fetch_model(self) -> Optional[AugmentedMSM]:
+        r""" Yields the most recently estimated AMM or None if :meth:`fit` was not called yet.
+
+        Returns
+        -------
+        amm : AugmentedMSM or None
+            The AMM instance.
+        """
+        return self._model
+
+    def fit(self, data, *args, **kw):
+        r""" Fits an AMM.
+
+        Parameters
+        ----------
+        data : TransitionCountModel or (N, N) ndarray
+            Count matrix over data.
+        *args
+            scikit-learn compatibility argument
+        **kw
+            scikit-learn compatibility argument
+
+        Returns
+        -------
+        self : AugmentedMSMEstimator
+            Reference to self.
+        """
+        if not isinstance(data, (TransitionCountModel, np.ndarray)):
+            raise ValueError("Can only fit on a TransitionCountModel or a count matrix directly.")
+
+        if isinstance(data, np.ndarray):
+            if data.ndim != 2 or data.shape[0] != data.shape[1] or np.any(data < 0.):
+                raise ValueError("If fitting a count matrix directly, only non-negative square matrices can be used.")
+            count_model = TransitionCountModel(data)
+        else:
+            count_model = data
+
+        if len(self.experimental_measurement_weights) != self.expectations_by_state.shape[1]:
+            raise ValueError("Experimental weights must span full observable space.")
+        if len(self.experimental_measurements) != self.expectations_by_state.shape[1]:
+            raise ValueError("Experimental measurements must span full observable state space.")
+
+        count_matrix = count_model.count_matrix
+        if issparse(count_matrix):
+            count_matrix = count_matrix.toarray()
+
+        # slice out active states from E matrix
+        expectations_selected = self.expectations_by_state[count_model.state_symbols]
+        count_matrix_symmetric = 0.5 * (count_matrix + count_matrix.T)
+        nonzero_counts = np.nonzero(count_matrix_symmetric)
+        counts_row_sums = np.sum(count_matrix, axis=1)
+        expectations_confidence_interval = confidence_interval(expectations_selected, conf=self.support_confidence)
+
+        measurements = self.experimental_measurements
+        measurement_weights = self.experimental_measurement_weights
+
+        count_outside = []
+        count_inside = []
+
+        i = 0
+        # Determine which experimental values are outside the support as defined by the Confidence interval
+        for emi, ema, mm, mw in zip(expectations_confidence_interval[0], expectations_confidence_interval[1],
```

Comment:
`emi, ema, mm, mw` is rough i advocate for a small `#comment` or something clearer